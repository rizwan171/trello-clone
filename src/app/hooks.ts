import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import type { RootState, AppDispatch } from "./store";

// The below useAppDispatch does not require a return type, hence the eslint disable comment
// See https://stackoverflow.com/questions/70711795/missing-return-type-on-function-useappdispatch for more info
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
