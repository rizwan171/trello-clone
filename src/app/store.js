import { configureStore } from "@reduxjs/toolkit";
import boardReducer from "../features/boardSlice.js";
import listsReducer from "../features/listsSlice.js";
import cardsReducer from "../features/cardsSlice.js";
import boardOptionsReducer from "../features/boardOptionSlice.js";
import currentSelectedCardReducer from "../features/currentSelectedCardSlice.js";
import currentSelectedListReducer from "../features/currentSelectedListSlice.js";
import listOptionsMenuPositionReducer from "../features/listOptionsMenuPositionSlice.js";
import tagsReducer from "../features/tagsSlice.js";
import selectedTagColourReducer from "../features/selectedTagColourSlice.js";
import createTagMenuDataReducer from "../features/createTagMenuDataSlice.js";
import modalActionMenusVisibilityReducer from "../features/modalActionMenusVisibilitySlice.js";

export const store = configureStore({
  reducer: {
    board: boardReducer,
    lists: listsReducer,
    cards: cardsReducer,
    boardOptions: boardOptionsReducer,
    currentSelectedCard: currentSelectedCardReducer,
    currentSelectedList: currentSelectedListReducer,
    listOptionsMenuPosition: listOptionsMenuPositionReducer,
    tags: tagsReducer,
    selectedTagColour: selectedTagColourReducer,
    createTagMenuData: createTagMenuDataReducer,
    modalActionMenusVisibility: modalActionMenusVisibilityReducer,
  },
});
