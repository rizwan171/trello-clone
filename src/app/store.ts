import { combineReducers, configureStore, PreloadedState } from "@reduxjs/toolkit";
import boardReducer from "../features/boardSlice";
import listsReducer from "../features/listsSlice";
import cardsReducer from "../features/cardsSlice";
import boardOptionsReducer from "../features/boardOptionSlice";
import currentSelectedCardReducer from "../features/currentSelectedCardSlice";
import currentSelectedListReducer from "../features/currentSelectedListSlice";
import listOptionsMenuPositionReducer from "../features/listOptionsMenuPositionSlice";
import tagsReducer from "../features/tagsSlice";
import selectedTagColourReducer from "../features/selectedTagColourSlice";
import createTagMenuDataReducer from "../features/createTagMenuDataSlice";
import modalActionMenusVisibilityReducer from "../features/modalActionMenusVisibilitySlice";

const rootReducer = combineReducers({
  board: boardReducer,
  lists: listsReducer,
  cards: cardsReducer,
  boardOptions: boardOptionsReducer,
  currentSelectedCard: currentSelectedCardReducer,
  currentSelectedList: currentSelectedListReducer,
  listOptionsMenuPosition: listOptionsMenuPositionReducer,
  tags: tagsReducer,
  selectedTagColour: selectedTagColourReducer,
  createTagMenuData: createTagMenuDataReducer,
  modalActionMenusVisibility: modalActionMenusVisibilityReducer,
});

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export type RootState = ReturnType<typeof rootReducer>
export type AppDispatch = AppStore['dispatch']
export type AppStore = ReturnType<typeof setupStore>