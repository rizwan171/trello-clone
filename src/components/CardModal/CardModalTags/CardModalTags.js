import React from "react";
import { useSelector } from "react-redux";
import AddTag from "../AddTag/AddTag";
import CardModalTag from "./CardModalTag/CardModalTag";

const CardModalTags = () => {
  const currentSelectedCard = useSelector((state) => state.currentSelectedCard.value);
  const tagsToShow = useSelector((state) => state.tags.value).filter((tag) => currentSelectedCard.tags.includes(tag.id));

  return (
    <div className="flex flex-col text-sm text-gray-700">
      <p className="mb-2">Tags</p>
      <div className="flex gap-1 flex-wrap mb-4">
        {tagsToShow.map((tag) => (
          <CardModalTag key={tag.id} tag={tag} />
        ))}
        <AddTag />
      </div>
    </div>
  );
};

export default CardModalTags;
