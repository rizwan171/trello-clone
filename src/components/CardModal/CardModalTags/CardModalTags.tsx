import { useAppSelector } from "../../../app/hooks";
import AddTag from "../AddTag/AddTag";
import CardModalTag from "./CardModalTag/CardModalTag";

const CardModalTags = () => {
  const card = useAppSelector((state) => state.currentSelectedCard.value);
  const tagsToShow = useAppSelector((state) => state.tags.value).filter((tag) => card?.tags.includes(tag.id));

  return (
    <div className="flex flex-col text-sm text-gray-700">
      <p className="mb-2">Tags</p>
      <div className="flex gap-1 flex-wrap mb-4">
        {tagsToShow.map((tag) => (
          <CardModalTag key={tag.id} tag={tag} />
        ))}
        <AddTag />
      </div>
    </div>
  );
};

export default CardModalTags;
