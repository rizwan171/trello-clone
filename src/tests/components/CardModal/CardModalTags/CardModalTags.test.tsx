import { RootState } from "../../../../app/store";
import CardModalTags from "../../../../components/CardModal/CardModalTags/CardModalTags";
import { renderWithProviders } from "../../../utils/renderUtils";

describe("CardModalTags", () => {
  let initialState: Partial<RootState>;

  beforeEach(() => {
    initialState = {
      currentSelectedCard: {
        value: {
          id: "1",
          title: "Test Card",
          description: "Test Description",
          listId: "1",
          attachments: [],
          tags: ["1", "2"],
        },
      },
      tags: {
        value: [
          {
            id: "1",
            name: "",
            colour: "#399e5a",
          },
          {
            id: "2",
            name: "",
            colour: "#5ABCB9",
          },
          {
            id: "3",
            name: "",
            colour: "#26532b",
          },
        ],
      },
    };
  });

  it("should render successfully", () => {
    const view = renderWithProviders(<CardModalTags />, { preloadedState: initialState });
    expect(view.asFragment()).toMatchSnapshot();
  });
});
