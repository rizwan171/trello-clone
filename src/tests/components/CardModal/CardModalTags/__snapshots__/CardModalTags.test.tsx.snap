// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardModalTags should render successfully 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col text-sm text-gray-700"
  >
    <p
      class="mb-2"
    >
      Tags
    </p>
    <div
      class="flex gap-1 flex-wrap mb-4"
    >
      <div
        class="flex items-center gap-1 py-1 px-2 rounded-sm font-semibold text-white"
        style="background: rgb(57, 158, 90);"
      >
        <p />
      </div>
      <div
        class="flex items-center gap-1 py-1 px-2 rounded-sm font-semibold text-white"
        style="background: rgb(90, 188, 185);"
      >
        <p />
      </div>
      <button
        class="p-1 rounded-ibsm flex items-center cursor-pointer bg-black bg-opacity-5 hover:bg-opacity-10"
      >
        <svg
          class="mr-0.5 text-trello-gray-200"
          fill="none"
          height="25"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="12"
            x2="12"
            y1="5"
            y2="19"
          />
          <line
            x1="5"
            x2="19"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
